### PR TITLE
Fixing filter when adding user groups at workspace

### PR DESCRIPTION
**Pull request by Leandroqo**
## Problem

In Workspace> Add user group: is not using the parameter to filter a search by name.

## Solution

- Added debounce to wait 0,5 seconds before making an API request.
- Added react-select/async to load options from an API call into the select component.

## Before & After Screenshots

**BEFORE**:
before

**AFTER**:
after

### CODE REVIEW

File: ui/src/modules/Settings/Credentials/Sections/UserGroup/__tests__/FormUserGroup.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  const addButton = await screen.findByText('Add');
+  expect(addButton).toBeInTheDocument();
+    
+  const selectUserGroup = screen.getByText('Select a user group');
+  await selectEvent.select(selectUserGroup, 'Maintainer');
+
+  expect(screen.getByTestId('button-default-add')).not.toBeDisabled();
+});
+
+test('should find a user group by name', async () => {
```
Comment (bCatanant) [inline ui/src/modules/Settings/Credentials/Sections/UserGroup/__tests__/FormUserGroup.spec.tsx:45]: Remove coments on mockResponseOnce's

Comment (luanecavalcantizup) [reply ui/src/modules/Settings/Credentials/Sections/UserGroup/__tests__/FormUserGroup.spec.tsx:45]: vdd, obrigado :)

**Review by bCatanant — APPROVED** [review]

**Review by bCatanant — APPROVED** [review]

File: ui/src/modules/Settings/Credentials/Sections/UserGroup/Form.tsx [hunk 2]
```diff
@@ -17,6 +17,7 @@
 import React, { useState, useEffect } from 'react';
 import map from 'lodash/map';
 import isEmpty from 'lodash/isEmpty';
+import debounce from 'debounce-promise';
```
Comment (ifdouglas) [inline ui/src/modules/Settings/Credentials/Sections/UserGroup/Form.tsx:20]: `lodash/debounce` does not attend?

**Review by ifdouglas — CHANGES_REQUESTED** [review]
### Maybe a behavior issue
- After selecting an UserGroup and then try change to another, I can not delete the filter unless I start typing.

**Review by ifdouglas — APPROVED** [review]

**Pull request merged**